coreneuron/io/lfp.hpp

```diff
@@ -0,0 +1,226 @@
+#ifndef LFP_H
+#define LFP_H
+
+#include <cmath>
+#include <array>
+#include <iostream>
+#include <vector>
+
+#include <boost/math/constants/constants.hpp>
+
+#include "coreneuron/utils/nrn_assert.h"
+#include "coreneuron/mpi/nrnmpidec.h"
+
+namespace coreneuron {
+
+template <typename T>
+using Array3 = std::array<T, 3>;
+
+template <typename Point3D, typename F>
```

Comment:
I am not sure here the template is doing what is supposed to do.
First it seems to me that the template in this method (and the followings) is not necessary (if we can assume that the input needs to be a `Point3D` object and consequently the output is always a `double`).
Instead if you want to keep the template implementation, then I think it should be enabled only for `std::array<F,3>` and for `std::is_floating_point<F>::value == true`.
[Example](https://godbolt.org/z/MzMjbc)

Reply:
In addition to the previous comment (with which I agree) there is also the fact that we implement everything for points in 3D. With just a few changes we could implement all these functions for points in n dimensions

Reply:
I will leave other dimensions for another PR if that is ok.